src/Symfony/Component/ExpressionLanguage/ParsedExpression.php

```diff
@@ -42,6 +42,17 @@ public function getNodes()
 
     public function dump()
     {
-        return $this->nodes->dump();
+        return $this->dumpNode($this->nodes);
+    }
+
+    protected function dumpNode(Node $node)
```

Comment:
True! I now made the method private. There is no need for any extension point: one should just call $expr->getNodes() then do things with the result.